ivre/xmlnmap.py

```diff
@@ -589,12 +597,31 @@ def change_smb_enum_shares(table):
     if not table:
         return table
     result = {}
-    for field in ["account_used", "note"]:
-        if field in table:
+    for field in list(table):
+        if field == 'shares':
+            continue
+        if not (field.startswith('\\\\')
+                and isinstance(table[field], dict)):
             result[field] = table.pop(field)
-    result["shares"] = [
-        dict(value, Share=key) for key, value in viewitems(table)
-    ]
+    if 'shares' in table:
+        # We just need to fix the share names
+        result["shares"] = [
+            dict(elt, Share=_smb_enum_shares_fix_share_name(elt["Share"]))
+            for elt in result["shares"]
+        ]
+    else:
+        # We need to update the structured output to avoid data in field names:
+        #
+        # Old:
+        # {"\\ServerName\ShareName": {XXX}, ...}
+        # New:
+        # [{"Share": "\\ServerName\ShareName", XXX}, ...]
+        result["shares"] = [
```

Comment:
There is (at least) one problem with this: since we cannot know the order of the numeration, we cannot know the order of the resulting list. An option would be to use a `sorted()` call to fix that:
```Python
        result["shares"] = sorted(
            (
                dict(
                    value,
                    Share=_smb_enum_shares_fix_share_name(key)
                ) for key, value in viewitems(table)
            ),
            key=lambda elt: elt["Share"],
        )
```
(I have not tested this code, but I think it should work)

What do you think?

Reply:
You are right. It works.
Thank you.